packages/pn-commons/src/components/__test__/CustomDatePicker.test.tsx

```diff
@@ -1,44 +1,71 @@
-import currentLocale from 'date-fns/locale/it';
 import React from 'react';
 
 import { TextField } from '@mui/material';
-import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
-import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 
-import { render } from '../../test-utils';
+import { RenderResult, fireEvent, render, screen, waitFor } from '../../test-utils';
 import CustomDatePicker from '../CustomDatePicker';
 
-const WrappedCustomDatePicker = () => {
-  return (
-    <LocalizationProvider
-      id="startDate"
-      name="startDate"
-      dateAdapter={AdapterDateFns}
-      adapterLocale={currentLocale}
-    >
-      <CustomDatePicker
-        label={'DatePicker'}
-        onChange={() => {}}
-        value={new Date()}
-        renderInput={(params) => (
-          <TextField
-            {...params}
-            inputProps={{
-              ...params.inputProps,
-              placeholder: 'datepickerinput',
-            }}
-          />
-        )}
+const RenderDatePicker = ({ language = 'it' }: { language?: string }) => (
+  <CustomDatePicker
+    label={'DatePicker'}
+    onChange={() => {}}
+    value={new Date('01/01/2023')}
+    language={language}
+    renderInput={(params) => (
+      <TextField
+        {...params}
+        inputProps={{
+          ...params.inputProps,
+          placeholder: 'datepickerinput',
+        }}
       />
-    </LocalizationProvider>
-  );
-};
-
+    )}
+  />
+);
 describe('test CustomDatePicker component', () => {
   it('renders the component', () => {
-    const { getByPlaceholderText, container } = render(<WrappedCustomDatePicker />);
+    const { getByPlaceholderText, container } = render(<RenderDatePicker />);
     const input = getByPlaceholderText(/datepickerinput/i);
     expect(container).toHaveTextContent(/datepicker/i);
     expect(input).toBeInTheDocument();
   });
 });
+
+describe('test CustomDatePicker languages', () => {
+  const languages = [
+    {
+      language: 'it',
+      month: 'gennaio',
+    },
+    {
+      language: 'en',
+      month: 'january',
+    },
+    {
+      language: 'fr',
+      month: 'janvier',
+    },
+    {
+      language: 'de',
+      month: 'januar',
+    },
+    {
+      language: 'sl',
+      month: 'januar',
+    },
+  ];
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+  });
+  languages.forEach((language) => {
```

Comment:
Standardize this loop with the others in the app. check GenericAppErrorFactory.test